src/Model/ParameterGo.cs

```diff
@@ -156,7 +156,32 @@ public string ValueForMap(bool useDefault = false)
             }
             else if (IsClientProperty)
             {
-                value = "client." + CodeNamerGo.Instance.GetPropertyName(Name.Value);
+                var propName = CodeNamerGo.Instance.GetPropertyName(Name.Value);
+                // verify that the calculated name matches the property name
+                bool found = false;
+                foreach (var clientProp in Method.CodeModel.Properties)
+                {
+                    // prefer an exact match
+                    if (clientProp.Name == propName)
+                    {
+                        found = true;
+                        break;
+                    }
+                }

```

Comment:
hmm, a little more inefficient but i dont think impactful:
```
                var propName = CodeNamerGo.Instance.GetPropertyName(Name.Value);

                // verify that the calculated name matches the property name
                bool found = Method.CodeModel.Properties.Any(clientProp => clientProp.Name == propName);

                // didn't find an exact match, find the closest match.  we hit this case if
                // the front-end decided to add a suffix (e.g. '1') to the client property name.
                if (!found)
                {
                    propName = Method.CodeModel.Properties.First(clientProp => System.Text.RegularExpressions.Regex.IsMatch(clientProp.Name, $"{propName}\\d")).Name;
                }
                value = $"client.{propName}";
```
also using regex to make sure we don't match by mistake another property ( i.e. "Prop1" , "PropFoo" and we are searching for "Prop")

Reply:
Good point on the regex, I've made this change.